src/Clara/cmd/core/poem/PoemHandler.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @file query handler for Poem Service for Remote type of connections for Sayori. 
+ * @author Capuccino
+ */
+
+const got = require('got');
+
+/**
+ * Simple handler for the [Sayori]{@link https://github.com/MonikaDesu/Sayori/} poem generator service.
+ */
+class PoemHandler {
+    /**
+     * Creates a new PoemHandler.
+     * 
+     * @param {String} [host='127.0.0.1'] Host which the poem server is running under.
+    */ 
+    constructor(host) {
+        host = host || 'http://127.0.0.1:8080';
+
+        if (typeof host !== 'string') throw new TypeError('host is not a string.');
```

Comment:
still needs that URL Regex so the lib knows its an actual URL so we don't cause errors hen calling the method. Having strict type check is the key

Reply:
IPs were failing with the regex. I'll find a way/different regex for validation.

Reply:
How about this 
```js
^(http(s?):\/\/)?(((www\.)?+[a-zA-Z0-9\.\-\_]+(\.[a-zA-Z]{2,3})+)|(\b(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\b))(\/[a-zA-Z0-9\_\-\s\.\/\?\%\#\&\=]*)?$```

[Citation](https://stackoverflow.com/questions/8249420/alter-regex-to-allow-ip-address-when-checking-url)